From 9bffeda1617cff031d31f497f50c6b4c4ecb35ad Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Mon, 24 Sep 2018 10:05:48 +0200
Subject: [PATCH] find-dbgsym-packages: Fix parsing [vdso: ...] lines in
 eu-unstrip output

Silences warnings similar to:

    Cannot parse eu-unstrip output: '0xf7f84000+0x2000 07b6d4113babfe81b7ef55f17d3afa875ab5a18c@0xf7f84570 . - [vdso: 6113]'

Fixes: 9a82dbc3e4f1540ca9f8a83fd9bebe5ab7ec38be
---
 find-dbgsym-packages | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/find-dbgsym-packages b/find-dbgsym-packages
index 318114d..4ebe310 100755
--- a/find-dbgsym-packages
+++ b/find-dbgsym-packages
@@ -89,16 +89,17 @@ sub parse_eu_unstrip
                       \s+
                       (\S+)
                       \s+
-                      (\S+)?
+                      (?: (\S+) | (\[vdso:\s+\d+\]) )?
                       $}ix) {
             my $id = $1;
             my $path = $2;
             my $debug = $3;
             my $name = $4 // $path;
+            my $vdso = $5;
             if ($debug ne '-') {
                 next;
             }
-            if ($name =~ /\[vdso: \d+\]/) {
+            if (defined $vdso) {
                 next;
             }
             if ($id eq '-') {
-- 
2.19.0

